### PR TITLE
actions: run tests on macos

### PR DESCRIPTION
**Pull request by oliver-sanders**
Bar the outstanding DNS issue and after a spate of portability changes Cylc Flow now supports MacOS nicely.

Add `macos-latest` into the CI tests:

* All unit and integration tests
  * Some of these required fixing for MacOS
  * Recent example of a portability test fix - https://github.com/cylc/cylc-flow/pull/3917
* The first 1/4 of functional tests
  * Running the whole test battery would be a bit much.

To sidestep the DNS issue this PR applies the [`cylc.flow.hostuserutil`](https://github.com/cylc/cylc-flow/issues/2689#issuecomment-715161888) I've been using for local development over the last year.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: .github/workflows/test_fast.yml [hunk 1]
```diff
@@ -18,7 +23,14 @@ jobs:
...
 
-      - name: Apt-Get
+      - name: Brew Install
+        if: startsWith(matrix.os, 'macos')
+        run: |
+          brew update
+          brew install shellcheck sqlite3
+
+      - name: Apt-Get Install
+        if: endsWith(matrix.os, 'ubuntu')
```
Comment (MetRonnie) [inline .github/workflows/test_fast.yml:33]: I think this is meant to be?
```suggestion
        if: startsWith(matrix.os, 'ubuntu')
```

File: .github/workflows/test_functional.yml [hunk 2]
```diff
@@ -37,7 +44,25 @@ jobs:
...
+
+          # add GNU coreutils and sed to the user PATH
+          # (see instructions in brew install output)
+          cat >> "$HOME/.bashrc" <<__HERE__
+          PATH="$(brew --prefix)/opt/coreutils/libexec/gnubin:$PATH"
+          PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"
+          __HERE__
+
+      - name: Apt-Get Install
+        if: endsWith(matrix.os, 'ubuntu')
```
Comment (MetRonnie) [inline .github/workflows/test_functional.yml:65]: Ditto
```suggestion
        if: startsWith(matrix.os, 'ubuntu')
```

File: .github/workflows/test_fast.yml (hunk 1 above)
Comment (oliver-sanders) [reply .github/workflows/test_fast.yml:33]: Crud, well spotted.

Comment (oliver-sanders) [reply .github/workflows/test_fast.yml:33]: Amusingly the tests all pass since the things we are installing are already in the base environment.

**Review by MetRonnie — APPROVED** [review]

**Review by kinow — APPROVED** [review]

**Pull request merged**